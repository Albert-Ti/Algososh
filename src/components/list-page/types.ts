export type TIsAction = {
  add: { [key: string]: boolean }
  remove: { [key: string]: boolean }
  state: { [key: string]: boolean }
}
