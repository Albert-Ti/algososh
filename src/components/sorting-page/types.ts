import { Direction } from '../../types/direction'
import { ElementStates } from '../../types/element-states'

export type TData = {
  items: TColumn[]
  load: boolean
}

export type TFilter = {
  sort: Direction | ''
  algorithm: 'bubble-sort' | 'selection-sort' | string
}

export type TColumn = {
  id: number
  column: number
  status: ElementStates
}

export type TAlgorithmStarted = {
  algorithm: 'bubble-sort' | 'selection-sort' | string
  start: boolean
  click: Direction | ''
}
