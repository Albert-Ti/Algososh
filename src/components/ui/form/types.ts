export type TForm = {
  onSubmit: (e: React.FormEvent<HTMLFormElement>) => void
  onChange: (value: string) => void
  inputValue: string | number
  loaderButton: boolean
  typeInput: 'text' | 'number'
  disabled: boolean
  textButton: string
}
