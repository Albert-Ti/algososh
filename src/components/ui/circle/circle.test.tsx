import { render } from '@testing-library/react'
import { Circle } from './circle'
import { ElementStates } from '../../../types/element-states'

describe('Testing the Circle Component', () => {
  it('correct rendering of an element without a letter', () => {
    const { container } = render(<Circle />)
    expect(container).toMatchSnapshot()
  })

  it('correct rendering of an element with a letter', () => {
    const { container } = render(<Circle letter='ABC' />)
    expect(container).toMatchSnapshot()
  })

  it('correct rendering of an element with head', () => {
    const { container } = render(<Circle head='1' />)
    expect(container).toMatchSnapshot()
  })

  it('correct rendering of an element with a react element in the head', () => {
    const { container } = render(<Circle head={<Circle />} />)
    expect(container).toMatchSnapshot()
  })

  it('correct rendering of an element with tail', () => {
    const { container } = render(<Circle tail='1' />)
    expect(container).toMatchSnapshot()
  })

  it('correct rendering of an element with a react element in the tail', () => {
    const { container } = render(<Circle tail={<Circle />} />)
    expect(container).toMatchSnapshot()
  })

  it('correct rendering of an element with index', () => {
    const { container } = render(<Circle index={0} />)
    expect(container).toMatchSnapshot()
  })

  it('correct rendering of an element with the property: isMail=true', () => {
    const { container } = render(<Circle isSmall={true} />)
    expect(container).toMatchSnapshot()
  })

  it('correct rendering of the element in the default state', () => {
    const { container } = render(<Circle state={ElementStates.Default} />)
    expect(container).toMatchSnapshot()
  })

  it('correct rendering of the element in the changing state', () => {
    const { container } = render(<Circle state={ElementStates.Changing} />)
    expect(container).toMatchSnapshot()
  })

  it('correct rendering of the element in the modified state', () => {
    const { container } = render(<Circle state={ElementStates.Modified} />)
    expect(container).toMatchSnapshot()
  })
})
