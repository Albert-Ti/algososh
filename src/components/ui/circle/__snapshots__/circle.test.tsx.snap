// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Circle Component correct rendering of an element with a letter 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        ABC
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Testing the Circle Component correct rendering of an element with a react element in the head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Testing the Circle Component correct rendering of an element with a react element in the tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing the Circle Component correct rendering of an element with head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      1
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Testing the Circle Component correct rendering of an element with index 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      0
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</div>
`;

exports[`Testing the Circle Component correct rendering of an element with tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      1
    </div>
  </div>
</div>
`;

exports[`Testing the Circle Component correct rendering of an element with the property: isMail=true 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Testing the Circle Component correct rendering of an element without a letter 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Testing the Circle Component correct rendering of the element in the changing state 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Testing the Circle Component correct rendering of the element in the default state 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Testing the Circle Component correct rendering of the element in the modified state 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;
